ios/rüm/DataViewController.swift

```diff
@@ -45,68 +48,105 @@ class DataViewController: UIViewController {
             return
         }
         
-        guard let members = userInfo["members"] as? [[String:AnyObject]] else {
+        guard let m = userInfo["members"] as? [[String:AnyObject]] else {
             return
         }
         
-        NSUserDefaults.standardUserDefaults().setValue(members, forKey: Constants.MEMBER_DATA)
+        NSUserDefaults.standardUserDefaults().setValue(m, forKey: Constants.MEMBER_DATA)
+        
         
-        self.showData(members)
+        self.members = m
+        dispatch_async(dispatch_get_main_queue(), {
+            self.collectionView.reloadData()
+            self.activityIndicator.hidden = true
+        })
+    }
+    
+    // MARK:- Collection view
+    func collectionView(collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return self.members.count
+    }
+    
+    func numberOfSectionsInCollectionView(collectionView: UICollectionView) -> Int {
+        return 1
     }
     
-    private func showData(members:[[String:AnyObject]] ) {
+    func collectionView(collectionView: UICollectionView, cellForItemAtIndexPath indexPath: NSIndexPath) -> UICollectionViewCell {
+        let kudosFaceCell = self.collectionView.dequeueReusableCellWithReuseIdentifier("kudosFaceCell", forIndexPath: indexPath) as! KudosFaceCell
+        kudosFaceCell.load(members[indexPath.row])
+        kudosFaceCell.kudosButtonDelegate = self
         
-        var dataEntries: [ChartDataEntry] = []
-        var titles:[String] = []
+        let loggedInUserId = NSUserDefaults.standardUserDefaults().stringForKey("ID")
+        if kudosFaceCell.kudosButton.userId == loggedInUserId {
+            kudosFaceCell.kudosButton.disabled = true
```

Comment:
Cells will get reused. Need to add an 

 `else { kudosFaceCell.kudosButton.disabled = false`